Cron/RetrieveReviews.php

```diff
@@ -50,40 +51,55 @@ public function __construct(
         $this->log = $logger;
         $this->variable = $variable;
 
-        $this->filePath = $this->getFeedUrl();
+        $this->apiKey = $this->getApiKey();
+        $this->locationId = $this->getLocationId();
     }
 
 
-    public function getFeedUrl(){
+    public function getApiKey(){
         return $this->storeManager->getStore()->getConfig(
-            'kiyoh_settings/general/feed_url'
+            'kiyoh_settings/general/api_key'
         );
     }
 
+    public function getLocationId(){
+        return $this->storeManager->getStore()->getConfig(
+            'kiyoh_settings/general/location_id'
+        );
+    }
+
+
     public function processAggregateScores() {
 
-        $this->curl->get($this->filePath);
-        $output = $this->curl->getBody();
+        if (!$this->getLocationId()) {
+            throw new \Magento\Framework\Exception\CronException(__('Location ID missing, please set your location ID.'));
+        }
+
+        if (!$this->getApiKey()) {
+            throw new \Magento\Framework\Exception\CronException(__('API key missing, please set your API key.'));
+        }
 
-        $doc = simplexml_load_string($output);
+        $this->curl->addHeader('X-Publication-Api-Token', $this->getApiKey());
+        $this->curl->get('https://www.kiyoh.com/v1/publication/review/external/location/statistics?locationId=' . $this->getLocationId());
 
-        unset($doc->{'reviews'});
+        $ouput = $this->curl->getBody();
+        $jsonOutput = json_decode($ouput);
 
         try {
-            if (isset($doc->errorCode)) {
-                throw new \Magento\Framework\Exception\CronException(__('Invalid xml format, code data missing.'));
+            if (isset($jsonOutput->errorCode)) {
+                throw new \Magento\Framework\Exception\CronException(__('Kiyoh API error: ' . $jsonOutput->errorCode . ': ' . $jsonOutput->detailedError[0]->message));
```

Comment:
You can rewrite this using the `__()` function to this:

```php
throw new CronException(
    __(
        'Kiyoh API error: %1: %2', 
        $jsonOutput->errorCode,
        $jsonOutput->detailedError[0]->message ?? null
    )
);
```

That `?? null` is there as a fallback if the message can't be found in the `$jsonOutput`